packages/runtime/runtime-utils/src/summarizerNodeContract.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { ISnapshotTree } from "@fluidframework/protocol-definitions";
+import {
+    IGraphNode,
+    ISummarizeInternalResult,
+    ISummarizerNode,
+    ISummarizerNodeConfig,
+    ISummarizerNodeWithGC,
+} from "@fluidframework/runtime-definitions";
+import { SummarizerNode } from "./summarizerNode";
+import { ReadAndParseBlob } from "./summarizerNodeUtils";
+import { SummarizerNodeWithGC } from "./summarizerNodeWithGc";
+
+export { ReadAndParseBlob } from "./summarizerNodeUtils";
+
+export interface ISummarizerNodeRootContract {
+    startSummary(referenceSequenceNumber: number, summaryLogger: ITelemetryLogger): void;
+    completeSummary(proposalHandle: string): void;
+    clearSummary(): void;
+    refreshLatestSummary(
+        proposalHandle: string | undefined,
+        getSnapshot: () => Promise<ISnapshotTree>,
+        readAndParseBlob: ReadAndParseBlob,
+        correlatedSummaryLogger: ITelemetryLogger,
+    ): Promise<void>;
+}
+
+export interface IRootSummarizerNode extends ISummarizerNode, ISummarizerNodeRootContract {}
+
+export interface IRootSummarizerNodeWithGC extends ISummarizerNodeWithGC, ISummarizerNodeRootContract {}
+
+export const createRootSummarizerNode = (
+    logger: ITelemetryLogger,
+    /** Summarize function */
+    summarizeInternalFn: (fullTree: boolean) => Promise<ISummarizeInternalResult>,
+    /** Sequence number of latest change to new node/subtree */
+    changeSequenceNumber: number,
+    /**
+     * Reference sequence number of last acked summary,
+     * or undefined if not loaded from summary.
+     */
+    referenceSequenceNumber: number | undefined,
+    config: ISummarizerNodeConfig = {},
+): IRootSummarizerNode => SummarizerNode.createRoot(
+        logger,
+        summarizeInternalFn,
+        changeSequenceNumber,
+        referenceSequenceNumber,
+        config,
+    );
+
+export const createRootSummarizerNodeWithGC = (
+    logger: ITelemetryLogger,
+    /** Summarize function */
```

Comment:
nit: Consider adding these to a comment block before the function that describes the function and has @param to describe the parameters. I think it makes makes it look cleaner.